### PR TITLE
Upgrade to Polymer 3

### LINKED ISSUE #541: DEPRECATED: HTML Imports
While running the application in chrome I could see the warning `[Deprecation] HTML Imports is deprecated and will be removed in M73, around March 2019. Please use ES modules instead. See https://www.chromestatus.com/features/5144752345317376 for more details.`

If this is going to be deprecated what is the future of this application?

### LINKED ISSUE #578: Upgrade to Polymer 3
[Polymer 3](https://polymer-library.polymer-project.org/3.0/docs/upgrade) switches from deprecated HTML Imports to ES Modules and more typical component structure. In order to keep the codebase moving forward and modernize the tools.

### PR DESCRIPTION
**Pull request by abraham**
Initial step in upgrading to Polymer 3.

- [x] Rewrites components from HTML to JS files
- [x] Remove bower components
- [x] Resolve nunchuck compilation
- [x] Resolve missing data
- [x] Find replacements for components without NPM version
- [x] Fix map
- [x] Test building
- [x] Fix service worker
- [x] Test deployment
- [x] Fix app update toast is not being displayed
- [x] Fix notification permissions are always being asked for on page load
- [x] Simplify deployment
- [ ] Write upgrade guide

Resolves #578 Resolves #541

### CODE REVIEW

File: src/elements/about-organizer-block.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        color: var(--secondary-text-color);
+      }
+
+      paper-button {
+        margin: 0;
+      }
+    </style>
+
+    <div class="container" layout="" horizontal="">
+      <div layout="" horizontal="" center-center="" flex="" hidden\$="[[viewport.isPhone]]">
```
Comment (ozasadnyy) [inline src/elements/about-organizer-block.js]:
```suggestion
      <div layout horizontal center-center flex hidden$="[[viewport.isPhone]]">
```
Lets make attributes clean as much as possible

File: package.json [hunk 2]
```diff
@@ -52,15 +76,25 @@
...
-  "private": true,
-  "pre-commit": "lint"
+  "pre-commit": "lint",
+  "resolutions": {
+    "inherits": "2.0.3",
+    "samsam": "1.1.3",
+    "supports-color": "3.1.2",
+    "type-detect": "1.0.0"
+  },
+  "contributors": [
```
Comment (ozasadnyy) [inline package.json]: Please add yourself here :)

File: src/elements/about-organizer-block.js [hunk 3]
```diff
@@ -46,15 +46,15 @@ class AboutOrganizerBlock extends ReduxMixin(PolymerElement) {
       }
     </style>
 
-    <div class="container" layout="" horizontal="">
-      <div layout="" horizontal="" center-center="" flex="" hidden\$="[[viewport.isPhone]]">
+    <div class="container" layout horizontal>
+      <div layout horizontal center-center flex hidden\$="[[viewport.isPhone]]">
```
Comment (ozasadnyy) [inline src/elements/about-organizer-block.js]: `hidden\$=""` is that a new way of handling properties or there should be `hidden$=`?

Comment (abraham) [reply src/elements/about-organizer-block.js]: No. `modulizer` escaped a bunch of `$`s and I haven't fixed them yet.

**Comment by abraham** [comment]
Took another pass at updating and everything seems to be working locally. There is some stuff left around the service-worker and compilation that needs to be finished.

**Comment by abraham** [comment]
The current `build:prod` script results in what I think are circular dependency errors. I also took a stab at switching to prpl-server but that's a big change and would probably require a lot of moving files around with gulp.

**Comment by abraham** [comment]
Polymer is not doing any further work on the CLI and is encouraging projects move to other tooling. I may take that route if it fixes the build issues on this branch.

> With HTML Imports deprecated and our current libraries having adopted JavaScript Modules, mainstream tools like Rollup and Webpack now work well for Web Components projects. We have therefore decided not to make further investments in the Polymer CLI and underlying tools, and we encourage developers to migrate to alternative toolchains. open-wc is a great place to start.

https://github.com/Polymer/project/blob/master/Roadmap.md

**Comment by abraham** [comment]
Rebased off of master and switched to using es-dev-server.  WIll need to build a rollup config next for production builds and do a lot of testing.

**Comment by abraham** [comment]
I have a rollup build sort of working but there isn't a nunjucks plugin for it that I could find and the replace plugin I've attempted to use doesn't support logic/loops and doesn't seem to apply to index.html. I've got another plugin I'm going to try.

**Comment by abraham** [comment]
I've got Rollup configured to compile the nunjucks template now with the exception of `index.html`. Working on a solution for that.

**Comment by abraham** [comment]
I have a rollup build that's doing most of the work that's needed. Still need to do some work around service workers and deploying and then I think this is ready for review.

**Comment by abraham** [comment]
Things are mostly working and are deployed to https://hoverboard-dev.firebaseapp.com/. There are issues with notifications and signing that still need to be addressed. There seems to be a minor performance improvement even with page bundles not currently being lazyloaded.

Results for Polymer 3 without lazyloading:
Screen Shot 2020-02-09 at 12 29 10

Result for Polymer 2 with lazyloading:
Screen Shot 2020-02-09 at 12 32 35

**Comment by abraham** [comment]
The last two issues I'm working on are:
- [ ] notification permissions are always being asked for
- [ ] on app update toast is not being displayed

**Comment by abraham** [comment]
This has been deployed to:
- https://hoverboard-dev.firebaseapp.com/
- https://devfedstwi.com/

**Comment by abraham** [comment]
@ozasadnyy If you have a minute I'd love another look to see if I broke anything obvious.

**Comment by ozasadnyy** [comment]
image
I believe there should be only 4 blocks

**Comment by ozasadnyy** [comment]
Not sure if it was working but if you go speakers -> speaker details -> session details and press back nothing happens to the dialog
image

File: README.md [hunk 4]
```diff
@@ -29,29 +29,9 @@ Our goal is to allow event organizers to set up a professional conference websit
...
-   * `npm run build`
-   * `npm run deploy`
-
-*NOTE:* By default command using configurations from `/configs/development.json`.
-To serve locally or deploy the production app use `npm run start:prod` and `npm run deploy:prod` respectively.
+
+:first_quarter_moon_with_face: Read the [Set Up Guide](/docs/tutorials/00-set-up.md).
+
 
 :book: Read the [Full Setup Guide](/docs/).
```
Comment (pearlbea) [inline README.md]: The names "Set Up Guide" and "Full Setup Guide" are confusing. Maybe rename this to "Documentation"?

**Pull request merged**